go.mod

```diff
@@ -59,20 +59,27 @@ require (
 	github.com/vincent-petithory/dataurl v0.0.0-20191104211930-d1553a71de50
 	golang.org/x/crypto v0.0.0-20200622213623-75b288015ac9
 	golang.org/x/sync v0.0.0-20201008141435-b3e1573b7520
-	golang.org/x/text v0.3.3 // indirect
 	golang.org/x/tools v0.0.0-20200914190812-8f9ed77dd8e5 // indirect
 	gopkg.in/gormigrate.v1 v1.6.0
 	gopkg.in/square/go-jose.v2 v2.3.1
 	gopkg.in/yaml.v2 v2.3.0
 	honnef.co/go/tools v0.0.1-2020.1.5 // indirect
-	k8s.io/api v0.18.5
-	k8s.io/apimachinery v0.18.5
+	k8s.io/api v0.19.0
+	k8s.io/apimachinery v0.19.0
 	k8s.io/client-go v12.0.0+incompatible
-	k8s.io/utils v0.0.0-20200603063816-c1c6865ac451 // indirect
-	sigs.k8s.io/controller-runtime v0.6.0
+	sigs.k8s.io/controller-runtime v0.6.2
 )
 
 replace (
 	github.com/metal3-io/baremetal-operator => github.com/openshift/baremetal-operator v0.0.0-20200715132148-0f91f62a41fe // Use OpenShift fork
-	k8s.io/client-go => k8s.io/client-go v0.18.2
+	github.com/metal3-io/cluster-api-provider-baremetal => github.com/openshift/cluster-api-provider-baremetal v0.0.0-20190821174549-a2a477909c1d
+	github.com/openshift/hive => github.com/dgoodwin/hive v0.0.0-20210121160047-23364b143670
+	github.com/openshift/library-go => github.com/openshift/library-go v0.0.0-20200918101923-1e4c94603efe
+	github.com/terraform-providers/terraform-provider-aws => github.com/openshift/terraform-provider-aws v1.60.1-0.20200630224953-76d1fb4e5699
+	github.com/terraform-providers/terraform-provider-azurerm => github.com/openshift/terraform-provider-azurerm v1.40.1-0.20200707062554-97ea089cc12a
+	github.com/terraform-providers/terraform-provider-ignition/v2 => github.com/community-terraform-providers/terraform-provider-ignition/v2 v2.1.0
+	k8s.io/client-go => k8s.io/client-go v0.19.0
+	sigs.k8s.io/cluster-api-provider-aws => github.com/openshift/cluster-api-provider-aws v0.2.1-0.20200506073438-9d49428ff837
+	sigs.k8s.io/cluster-api-provider-azure => github.com/openshift/cluster-api-provider-azure v0.1.0-alpha.3.0.20200120114645-8a9592f1f87b
+	sigs.k8s.io/cluster-api-provider-openstack => github.com/openshift/cluster-api-provider-openstack v0.0.0-20200526112135-319a35b2e38e
```

Comment:
Is there a way to not have to maintain all of these versions in our app?
This feels similar to the problem I was trying to avoid in https://github.com/openshift/installer/issues/4330

I understand that this is probable more necessary than installconfig validation, but this all feels very fragile. What happens if we forget to update one of these refs?

Sorry for commenting after merge, but I found this because it seems we need go 1.15 for something related to this line:
`github.com/openshift/hive => github.com/dgoodwin/hive v0.0.0-20210121160047-23364b143670`

Reply:
Our build container is already in golang v1.15 the real issue is probably with the local IDE that is working with a different golang version. 
the line related to `dgoodwin` will be removed once his PR is pushed to hive.
About the other replace statements i didn't find a way not to have them, hive have them as well, do you know how to avoid it?

Reply:
Not without splitting the API we require into a separate module or sub-module as suggested in https://github.com/openshift/installer/issues/4330

Reply:
Actually it's a good idea, but they produce the same binary so how does it work?

Reply:
It would need a change on the hive side so that we could require just the apis and not all of the runtime dependencies.

Reply:
Lets talk about it in our next meeting